serialize/serialize.go

```diff
@@ -65,12 +66,23 @@ func ToBytes(tracer opentracing.Tracer, spanContext opentracing.SpanContext, req
 
 	var writer bytes.Buffer
 	err = protocol.Binary.Encode(wireValue, &writer)
-	return writer.Bytes(), err
+
+	// prepend single byte to version the serialization
+	// '0' indicates:
+	// 	thrift serialization (request) + jaeger.binary format (ctx/tracing)
+	yarpcBytes := append([]byte{0}, writer.Bytes()...)
```

Comment:
Instead of this, do [`WriteByte`](https://golang.org/pkg/bytes/#Buffer.WriteByte) on the buffer before calling encode.

Also, use an unexported constant.

```go
var writer bytes.Buffer
if err := writer.WriteByte(version); err != nil {
	return err
}
err = protocol.Binary.Encode(wireValue, &writer)
return writer.Bytes(), err
```